File: dogpack-developer/apps/2d/euler/ShockDiffraction/plotq2_cart.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%
%%   Information you have available to you:
%%
%%     Basic information:
%%                  mx, my:  number of points in each coordinate direction
%% [xlow,xhigh,ylow,yhigh]:  min/max values of grid
%%                    meqn:  number of equations
%%                    maux:  number of aux components
%%                   meth1:  spatial order of accuracy
%%
%%   Grid information:
%%       (xc,yc): grid points (cell centers), size = (mx,my)
%%       (xl,yl): grid points (lower left cell corners), size = (mx+1,my+1)
%%
%%   Solution information:
%%         qsoln:  solution sampled on mesh, size = (mx,my,meqn)
%%           aux:  aux components sampled on mesh, size = (mx,my,maux)
%%          qaug:  solution sampled on mesh, with zero padding to
%%                 make it compatible with surf and pcolor matlab
%%                 plotting tools, size = (mx+1,my+1,meqn)
%%       aux_aug:  aux components sampled on mesh, with zero padding to
%%                 make it compatible with surf and pcolor matlab
%%                 plotting tools, size = (mx+1,my+1,maux)
%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% gas constant
fids  = fopen([outputdir,'/eulerhelp.dat'],'r');
if fids==-1
  disp(['File  ',outputdir,'/eulerhelp.dat  not found.']);
  disp('Setting gamma = 1.4');
  gamma_gas = 1.4;
else
  gamma_gas  = fscanf(fids,'%e',1);
  fclose(fids);
end

figure(1);
clf;
rho=qsoln(:,:,m);
rho(xc<=1 & yc<=6)=0.0;
pcolor(xc,yc,rho);
shading flat;
yrbcolormap
axis on; box on; grid off;
axis('equal');
axis([-0.05 13.25 -0.05 11.05]);
set(gca,'xtick',-4:1.0:14);
set(gca,'ytick',-4:1.0:14);
set(gca,'fontsize',16);
t1 = title(['Density at t = ',num2str(time),'     [DOGPACK]']); 
set(t1,'fontsize',16);
colorbar;
%caxis([0.8,10]);
caxis auto;
   
% Grey scale figure plot that you see in a lot of papers.
figure(2);
clf;
contourf(xc, yc, rho, linspace(0.06627, 7.16, 20), '-k' );
colorbar()
axis on; box on; grid off;
axis('equal');
axis([-0.05 13.25 -0.05 11.05]);
set(gca,'xtick',-4:1.0:14);
set(gca,'ytick',-4:1.0:14);
set(gca,'fontsize',16);
t1 = title(['Density at t = ',num2str(time),'     [DOGPACK]']); 
set(t1,'fontsize',16);

figure(3)
clf;
rho=qsoln(:,:,1);
e=qsoln(:,:,5);
u1=qsoln(:,:,2)./qsoln(:,:,1);
u2=qsoln(:,:,3)./qsoln(:,:,1);
u3=qsoln(:,:,4)./qsoln(:,:,1);
P=0.4*(e-0.5*rho.*(u1.*u1+u2.*u2+u3.*u3));
P(xc<=1 & yc<=6)=0.0;
c=sqrt(1.4*P./rho);
M=sqrt(u1.*u1+u2.*u2)./c;
%contour(xc, yc, P, linspace(0.091, 38, 50), '-k' );
contourf(xc, yc, P, linspace(0.091, 37, 40), '-k' );
colorbar()
axis on; box on; grid off;
axis('equal');
axis([-0.05 13.25 -0.05 11.05]);
set(gca,'xtick',-4:1.0:14);
set(gca,'ytick',-4:1.0:14);
set(gca,'fontsize',16);
t1 = title(['Pressure at t = ',num2str(time),'     [DOGPACK]']); 
set(t1,'fontsize',16);

% n1 = frame number
%fname = strcat( strcat( 'density', num2str(n1, '%02d' ) ), '.jpg' );
%print(1, '-djpeg', fname );
%fname = strcat( strcat( 'density-contour', num2str(n1, '%02d' ) ), '.eps' );
%print(3, '-deps', fname  );

File: dogpack-developer/apps/2d/euler/ShockDiffraction/plotq2_cart.py

#----------------------------------------------------------
def plotq2_cart(outputdir,nframe,
                m,meth1,meqn,time,
                points_per_dir,LegVals,
                xlow,xhigh,ylow,yhigh,
                mx,my,
                dx,dy,
                mx_old,my_old,
                dx_old,dy_old,
                xc,yc,
                xl,yl,
                qaug):
    
    import matplotlib.pyplot as plt
    import numpy as np
    qsoln=qaug;
    # HACK:
    gamma = 1.4
    gp1 = gamma+1.
    gm1 = gamma-1.
    print ('hi')
    # DENSITY
    if 1:
     m=0
     fig=plt.figure(1)
     plt.clf()
     plt.gca().set_aspect('auto')
     #plt.gca().set_xlim([xc[0],xc[mx-1]])
     #plt.gca().set_ylim([0.0, 7.0])
     colors=np.linspace(0.0662,7.07,20)
     plt.contour(xl,yl,qsoln[:,:,m],colors)
     xo=np.linspace(0.0,1.0,40)
     yo1=0.0+0*xo
     yo2=6.0+0*xo
     plt.plot(xo,yo1,xo,yo2)
     plt.fill_between(xo,yo1,yo2, color='grey',zorder=3)
     tmp1 = "".join(("Density at t = ",str(time)))
     title = "".join((tmp1,"     [DoGPack]"))
     plt.title(title)
     fig.tight_layout()
     plt.savefig('rarefaction_density.pdf')


    # PRESSURE
    if 1:
     m=1
     press = (gamma-1)*(qsoln[:,:,4]-0.5/qsoln[:,:,0]*(
        qsoln[:,:,1]**2 + qsoln[:,:,2]**2 + qsoln[:,:,3]**2))
     print press[press != 0].min()
     fig=plt.figure(2)
     plt.clf()
     plt.gca().set_aspect('auto')
     #plt.gca().set_xlim([xc[0],xc[mx-1]])
     #plt.gca().set_ylim( [0.0, 0.2] )
     colors=np.linspace(0.091,37,38)
     plt.contour(xl,yl,press,colors)
     xo=np.linspace(0.0,1.0,40)
     yo1=0.0+0*xo
     yo2=6.0+0*xo
     plt.plot(xo,yo1,xo,yo2)
     plt.fill_between(xo,yo1,yo2, color='grey',zorder=3)
     tmp1 = "".join(("Pressure at t = ",str(time)))
     title = "".join((tmp1,"     [DoGPack]"))
     plt.title(title)
     fig.tight_layout()
     plt.savefig('rarefaction_pressure.pdf')

 
    # Velocity
    if 1:
     m=2
     u = np.zeros(mx,float)
     u = qsoln[:,:,1] #/ qsoln[:,:,0]
        
     fig=plt.figure(3)
     plt.clf()
     plt.gca().set_aspect('auto')
     #plt.gca().set_xlim([xc[0],xc[mx-1]])
     #plt.gca().set_ylim([-1.0, 1.0])
     colors=np.linspace(0.091,37,40)
     plt.contourf(xl,yl,u, 20)
     tmp1 = "".join(("Velocity at t = ",str(time)))
     title = "".join((tmp1,"     [DoGPack]"))
     plt.title(title)
     fig.tight_layout()
     #plt.savefig('rarefaction_velocity.pdf')
    
#----------------------------------------------------------
